imap/append.c

```diff
@@ -865,6 +865,33 @@ static int append_apply_flags(struct appendstate *as,
     return r;
 }
 
+struct findstage_cb_rock {
+    const char *partition;
+    char *stagefile;
+    size_t size;
+};
+
+static int findstage_cb(const conv_guidrec_t *rec, void *vrock)
+{
+    struct findstage_cb_rock *rock = vrock;
+    mbentry_t *mbentry = NULL;
+
+    int r = mboxlist_lookup(rec->mboxname, &mbentry, NULL);
+    if (r) return 0;
+
+    if (!strcmp(rock->partition, mbentry->partition)) {
+        strlcpy(rock->stagefile,
+                mboxname_datapath(mbentry->partition, mbentry->name,
+                                  mbentry->uniqueid, rec->uid),
+                rock->size);
+        r = CYRUSDB_DONE;
+    }
```

Comment:
I think we should be hard-linking the file into the spool path rather than setting the stagefile to be directly the path inside the mailbox.